vault/config.go

```diff
@@ -138,20 +140,26 @@ func (c *ConfigFile) parseFile() error {
 
 // ProfileSection is a profile section of the config file
 type ProfileSection struct {
-	Name            string `ini:"-"`
-	MfaSerial       string `ini:"mfa_serial,omitempty"`
-	RoleARN         string `ini:"role_arn,omitempty"`
-	ExternalID      string `ini:"external_id,omitempty"`
-	Region          string `ini:"region,omitempty"`
-	RoleSessionName string `ini:"role_session_name,omitempty"`
-	DurationSeconds uint   `ini:"duration_seconds,omitempty"`
-	SourceProfile   string `ini:"source_profile,omitempty"`
-	ParentProfile   string `ini:"parent_profile,omitempty"` // deprecated
-	IncludeProfile  string `ini:"include_profile,omitempty"`
-	SSOStartURL     string `ini:"sso_start_url,omitempty"`
-	SSORegion       string `ini:"sso_region,omitempty"`
-	SSOAccountID    string `ini:"sso_account_id,omitempty"`
-	SSORoleName     string `ini:"sso_role_name,omitempty"`
+	Name                 string `ini:"-"`
+	MfaSerial            string `ini:"mfa_serial,omitempty"`
+	RoleARN              string `ini:"role_arn,omitempty"`
+	ExternalID           string `ini:"external_id,omitempty"`
+	Region               string `ini:"region,omitempty"`
+	RoleSessionName      string `ini:"role_session_name,omitempty"`
+	DurationSeconds      uint   `ini:"duration_seconds,omitempty"`
+	SourceProfile        string `ini:"source_profile,omitempty"`
+	ParentProfile        string `ini:"parent_profile,omitempty"` // deprecated
+	IncludeProfile       string `ini:"include_profile,omitempty"`
+	SSOStartURL          string `ini:"sso_start_url,omitempty"`
+	SSORegion            string `ini:"sso_region,omitempty"`
+	SSOAccountID         string `ini:"sso_account_id,omitempty"`
+	SSORoleName          string `ini:"sso_role_name,omitempty"`
+	AWSVaultBackend      string `ini:"aws_vault_backend,omitempty"`
+	AWSVaultPrompt       string `ini:"aws_vault_prompt,omitempty"`
+	AWSVaultKeychainName string `ini:"aws_vault_keychain_name,omitempty"`
+	AWSVaultPassCmd      string `ini:"aws_vault_pass_cmd,omitempty"`
+	AWSVaultPassDir      string `ini:"aws_vault_pass_dir,omitempty"`
+	AWSVaultPassPrefix   string `ini:"aws_vault_pass_prefix,omitempty"`
```

Comment:
I'm not keen to allow global config changes per profile, this would add a new layer of complexity with different profiles using different backends altogether 😱 . 

Should we consider splitting this config out, perhaps use the `ini` library directly to get this instead of `vault/config`. And maybe we should consider be using the `plugin` feature of aws cli to store the config in a different section name altogether (rather than under the default profile)?

What are your thoughts?

Reply:
Actually I also prefer not to store keyring parameters in aws profile config
Could you look, please, on #590 ?